### DISCUSSION TITLE
Is there a way to define a limit on how many times an adversarial image is generated using signOPT?

### DISCUSSION THREAD

**Discussion by teenaxta** [Trusted-AI/adversarial-robustness-toolbox#2100] [Q&A]
Hello

I have written a custom function and now I am attacking it using signOPT. The issue is that while it works great for some image pairs, for other image pairs it keeps on going, i ran for a good 10 mins but nothing happened. 

```
classifier = BlackBoxClassifier(self.comparison_function, self.im1.shape, 3, clip_values=(0, 255) )
attack = SignOPTAttack(estimator=classifier, targeted= True, max_iter=1,query_limit=1, num_trial=1, k=1)
self.x_adv = attack. Generate(x = np.array([self.im1],dtype=np.float32), x_init = np.array([self.im2],dtype=np.float32), y = to_categorical([1], 3), resume = False)
```
I have set all limits like max iter, query limit, num_trial to 1 but nothing changes, it still keeps on going forever.

**Comment by beat-buesser** [comment]
Hi @teenaxta Why did you set all the parameters to 1?

**Reply by teenaxta** [reply]
i am trying to limit the time it takes to come up with an adversarial image. right now for some images it takes 10 -20 mins to get the adversarial image. so thats why i have set everything to 1

**Reply by beat-buesser** [reply]
This limits the attack, for example with `num_trial=1` it can only do one attempt to find a random adversarial starting point and if that is not successful it will not return an adversarial example. Try to increase the values towards the default values which should be close to the values used by the authors of SignOPT.
